### ISSUE TITLE
ExportTypescript is  is too large error.

### ISSUE THREAD

**Issue by heaven-born** [scala-tsi/scala-tsi#304]
Getting this error `Generated bytecode for method 'com.scalatsi.generator.ExportTypescript$.<clinit>' is too large. Size: 68686 bytes. Limit is 64KB` during code generation. 

How is there a fix of workaround for this? 

I have one enum that contains 35 elements, so probably this is why...

**Comment by dhoepelman** [comment]
Interesting. Scala 2 or 3? I assume you don't have a public example?

This could be legitimate - I did not know about this limit -  if you "just" have a large number of exports and/or deep tree, or it could indicate a bug.

After you run `sbt typescriptCreateExporter`, do you see anything special in `target/scala-2.13/src_managed/com/scalatsi/generator/ExportTypescript.scala`?

**Comment by heaven-born** [comment]
this is the public example: https://github.com/heaven-born/ScalaTsiError

**Comment by dhoepelman** [comment]
Thank you for the example.

The Scala 3 codegen is fairly new and uses the `derived` mechanism. It might be that the generated code is too large, but this would require a deep-dive.

Another way to tackle this is to improve [the exporter application template](https://github.com/scala-tsi/scala-tsi/blob/master/plugin/src/main/twirl/ExportTypescriptTemplate.scala.txt#L15), where the too-large-bytecode constructor is generated.
I think this will have a small but acceptable performance decrease so it better than failing in this case.

Instead of the current code

```scala
val toOutput: Map[String, TypescriptType] = Map[String, TypescriptType](
    """Class1""" -> TSType.getOrGenerate[Class1].get,
    """Class2""" -> TSType.getOrGenerate[Class2].get,
    ...
)
```

That needs to be split up into something that gets make into different methods by the Scala compiler. I think this should work:

```scala
def `tstype_Class1` = TSType.getOrGenerate[Class1].get
def `tstype_Class2` = TSType.getOrGenerate[Class2].get

val toOutput: Map[String, TypescriptType] = Map[String, TypescriptType](
    """Class1""" -> `tstype_Class1`,
    """Class2""" -> `tstype_Class2`,
    ...
)
```

I do not know at what timeline I will get to making this change, but feel free to submit a PR

**Comment by dhoepelman** [comment]
Releasing fix in 0.8.4, thanks @hagl
